lib/OpenQA/Task/Needle/RemoveVersions.pm

```diff
@@ -0,0 +1,38 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Task::Needle::RemoveVersions;
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+
+use File::Find;
+use File::stat;
+
+my $minimum_retention_time;
+
+sub register ($self, $app, $job) {
+    $minimum_retention_time = $app->config->{'scm git'}->{minimum_needle_retention_time} * 60 // 30 * 60;
+    $app->minion->add_task(remove_needle_versions => sub ($job) { _remove_needle_versions($app, $job) });
+}
+
+sub _remove_needle_versions ($app, $job) {
```

Comment:
Could you add a unit test for this? At least something basic since this is a fair bit of new logic.

Btw it would also be good to mention this in the PR title

Reply:
I've had a go at adding a unit test for this in `t/14-grutasks.t`, I hope that's sufficient for now as this behaviour won't be active by default anyway.